docs/lib/snippets/modular/many_to_many/relational.drift.dart

```diff
@@ -597,4 +600,48 @@ class $$ShoppingCartEntriesTableTableManager extends i0.RootTableManager<
                   item: item,
                   rowid: rowid,
                 )));
+  $$ShoppingCartEntriesTableReferenceReader withReferences() {
+    return $$ShoppingCartEntriesTableReferenceReader(this);
+  }
+}
+
+class $$ShoppingCartEntriesTableReferenceReader<T0, T1>
+    extends i0.ReferenceReader<
+        i2.ShoppingCartEntry,
+        ({
+          i2.ShoppingCartEntry shoppingCartEntry,
+          T0? shoppingCart,
+          T1? item
+        })> {
+  $$ShoppingCartEntriesTableReferenceReader(this.$manager);
+  i0.GeneratedDatabase get _db => $manager.$state.db as i0.GeneratedDatabase;
+  final i0.BaseTableManager $manager;
+  @override
+  Future<({i2.ShoppingCartEntry shoppingCartEntry, T0? shoppingCart, T1? item})>
+      $withReferences(i2.ShoppingCartEntry value) async {
+    return (
```

Comment:
Does this effectively run two additional queries for each row in the main one? I think we should check whether we can transform the query to be a join so that we get the correct rows right away.

As it stands, this has a potential for race conditions (a query should always return a consistent snapshot between all tables involved). We could solve that by wrapping the whole thing in a transaction, but it's still quite inefficient and I think we should try to use a single statement with joins.

Reply:
Yeah, this part was done kind of quickly, so it's poorly optimized.  
The solution would be to either do a join, or doing a single 2nd query  `categories.filter((f)=>f.id.isIn([ ... ]))` to return a list of categories, using 1 transaction to avoid race conditions.

I'm still researching ways to have nested work, although I don't think there is way that would be both type-safe & easy to use.